### PR TITLE
Generate pdf template

### PR DESCRIPTION
**Pull request by MiZiet**

### CODE REVIEW

File: packages/generate-EN10168-pdf-template/package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "parcel-bundler": "^1.12.4",
+    "pdfmake": "^0.1.68",
+    "ts-node": "^9.0.0",
+    "typescript": "^4.0.5"
+  },
+  "scripts": {
+    "test": "mocha -r ts-node/register test/**/*.ts",
+    "build": "parcel build src/generateContent.ts --no-cache --no-source-maps",
+    "create-example": "ts-node utils/createExample.ts"
+  },
```
Comment (getlarge) [inline packages/generate-EN10168-pdf-template/package.json:21]: Could you inspire from the other packages for running the tests ?

Also could you add a script to trigger `update-version.ts` in order to have a fresh `schemaTypes.ts`

File: packages/generate-EN10168-pdf-template/test/renderingCommTrans.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import { expect } from 'chai';
+
+const certificate = require('./testingCertificate.json');
```
Comment (getlarge) [inline packages/generate-EN10168-pdf-template/test/renderingCommTrans.ts:3]: Please use the resources in fixtures folder, v0.0.2 is the version you build the pdf-template generator for.

File: packages/generate-EN10168-pdf-template/utils/DE.json [hunk 3]
```diff
@@ -0,0 +1,100 @@
+{
+  "certificateFields": {
+    "A01": "Herstellerwerk",
```
Comment (getlarge) [inline packages/generate-EN10168-pdf-template/utils/DE.json:3]: Please remove those translations

File: packages/generate-EN10168-pdf-template/utils/translate.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  this.languages = _languages;
+  this.translation = { FR, EN, DE, PL };
+  }
+
+  translate(phrase: string, group: 'certificateFields' | 'certificateGroups' | 'otherFields'): string{
+    if (group === 'certificateFields') return `${phrase} ${this.languages.map(language =>
+      this.translation[language][group][phrase]).join(' / ')}`;
+    return this.languages.map(language => this.translation[language][group][phrase]).join(' / ')
+  }
+}
```
Comment (getlarge) [inline packages/generate-EN10168-pdf-template/utils/translate.ts:22]: Here the idea is load the translations from the schema repository.

Please have a look at this function in generate-html module
```ts
const getTranslations = async (certificateLanguages: string[], schemaConfig: SchemaConfig): Promise<Translations> => {
  const translationsArray = await Promise.all(
    certificateLanguages.map(async (lang) => {
      const filePath = getRefSchemaUrl(schemaConfig, `${lang}.json`).href;
      return { [lang]: (await loadExternalFile(filePath, 'json')) as any };
    }),
  );

  return translationsArray.reduce((acc, translation) => {
    const [key] = Object.keys(translation);
    acc[key] = translation[key];
    return acc;
  }, {});
};
```

**Review by getlarge — CHANGES_REQUESTED** [review]
Looks pretty good! 
I will give it a try a bit later. In the meantime could you also remove yarn.lock ?
In this project i'm using package-lock.json files with lerna.

File: packages/generate-EN10168-pdf-template/utils/createExample.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import PdfPrinter from 'pdfmake';
+import * as fs from 'fs';
+import http from 'http';
+
+import { Translate } from './translate'
+
+import { styles } from './styles';
+
+import {generateContent} from '../dist/generateContent';
+// const file = fs.createWriteStream('fonts2/Lato-Regular.ttf');
```
Comment (krzysztof-jelski) [inline packages/generate-EN10168-pdf-template/utils/createExample.ts:10]: Remove the comment

File: packages/generate-EN10168-pdf-template/test/renderingSuppInform.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  it('correctly renders supplementary information for colSpan = 4', () => {
+    const suppInfo = {
+      "A11": {
+        "Key": "First Supplementary Information Commercial Transaction",
+        "Value": "1.0",
+        "Unit": "Apples"
+      },
+    }
+    const i18n = new Translate(['EN', 'DE']);
+    const supplementarInforamtion = supplementaryInformation(suppInfo, i18n, 4);
```
Comment (krzysztof-jelski) [inline packages/generate-EN10168-pdf-template/test/renderingSuppInform.ts:71]:
```suggestion
    const supplementarInformation = supplementaryInformation(suppInfo, i18n, 4);
```

File: packages/generate-EN10168-pdf-template/src/lib/supplementaryInformation.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { CommercialTransactionSupplementaryInformation, Translate } from '../types'
+
+const createBrackets = (amount: number) => [...Array(amount).fill({})];
```
Comment (krzysztof-jelski) [inline packages/generate-EN10168-pdf-template/src/lib/supplementaryInformation.ts:3]: Maybe we could name it after the goal/result. `createEmptyObjects`?

File: packages/generate-EN10168-pdf-template/src/lib/createProductDescription.ts [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+import { productNorms } from './productNorms';
+import { productShape} from './productShape';
+import {measurement} from './measurement';
+
+export function createProductDescription(productDescription: ProductDescription, i18n: Translate) {
+
+  const B02ProductNorms = productNorms(productDescription.B02, i18n);
+
+  const contentToOmit = ['B01', 'B02', 'B09', 'B10', 'B11', 'B12', 'B12','B13', 'SupplementaryInformation'];
+  const content = Object.keys(productDescription).filter(element => !contentToOmit.includes(element)).map(element =>
```
Comment (krzysztof-jelski) [inline packages/generate-EN10168-pdf-template/src/lib/createProductDescription.ts:12]: That's almost identical to how we generate `content` in `commercialTransaction`. Perhaps worth extracting a function.

File: packages/generate-EN10168-pdf-template/src/lib/createTransactionParties.ts [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      text: `${commercialTransaction[element].City},${commercialTransaction[element].ZipCode},${commercialTransaction[element].Country}`,
+      style: 'p'
+    }, { text: commercialTransaction[element].VAT_Id, style: 'p' }, { text: commercialTransaction[element].Email, style: 'p' }]
+  );
+
+  return [[...initKeys, ...keys], [...initValues, ...values]];
+}
+
+function fillTableRow(arr) {
+  if (arr.length === 3){
```
Comment (krzysztof-jelski) [inline packages/generate-EN10168-pdf-template/src/lib/createTransactionParties.ts:22]: The `3` in this file is worth extracting to a well-named constant.

**Review by getlarge — APPROVED** [review]

**Pull request merged**